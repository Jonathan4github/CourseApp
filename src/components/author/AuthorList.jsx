import React from 'react';
import PropTypes from 'prop-types';
import AuthorListRow from './AuthorListRow';


const propTypes = {
  authors: PropTypes.array.isRequired,
  handleDelete: PropTypes.func.isRequired
};

const AuthorList = ({ authors, handleDelete }) => (
  <table className="table table-hover">
    <thead>
      <tr>
        <th>ID</th>
        <th>First Name</th>
        <th>Last Name</th>
        <th>Action</th>
      </tr>
    </thead>
    <tbody>
      {authors
        .map(author => (
          <AuthorListRow
            key={author.id}
            author={author}
            handleDelete={handleDelete} />
        ))
      }
    </tbody>
  </table>
);

AuthorList.propTypes = propTypes;

export default AuthorList;
