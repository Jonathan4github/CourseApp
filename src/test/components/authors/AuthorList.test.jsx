/* eslint-disable */
import AuthorList from '../../../components/author/AuthorList';
import authorsMock from '../../__mocks__/authorMock';


describe('AuthorList component', () => {
  const setup = () => {
    const props = {
      authors: authorsMock.authors,
      handleDelete: jest.fn()
    };

    return {
      shallowWrapper: shallow(<AuthorList {...props} />),
      props
    };
  };


  it('should render without crashing', () => {
    const { shallowWrapper } = setup();

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('should check that proper table element are rendered', () => {
    const { shallowWrapper, props } = setup();

    expect(shallowWrapper.find('table').children()).toHaveLength(2);
    expect(shallowWrapper.find('tr').children()).toHaveLength(4);
    expect(shallowWrapper.find('AuthorListRow')).toHaveLength(props.authors.length);
  });
});
