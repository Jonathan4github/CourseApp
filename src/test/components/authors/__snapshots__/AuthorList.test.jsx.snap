// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorList component should render without crashing 1`] = `
<table
  className="table table-hover"
>
  <thead>
    <tr>
      <th>
        ID
      </th>
      <th>
        First Name
      </th>
      <th>
        Last Name
      </th>
      <th>
        Action
      </th>
    </tr>
  </thead>
  <tbody>
    <AuthorListRow
      author={
        Object {
          "firstName": "Cory",
          "id": "cory-house",
          "lastName": "House",
        }
      }
      handleDelete={[MockFunction]}
      key="cory-house"
    />
    <AuthorListRow
      author={
        Object {
          "firstName": "Scott",
          "id": "scott-allen",
          "lastName": "Allen",
        }
      }
      handleDelete={[MockFunction]}
      key="scott-allen"
    />
  </tbody>
</table>
`;
